docs/rules.md

```diff
@@ -57,9 +57,10 @@ Rules are sorted by severity.
 | `MOZILLA_COND_OF_USE`      | notice   | Mozilla conditions of use violation.                |
 | `FLAGGED_FILE_TYPE`        | notice   | (Binary) Flagged file type found.                   |
 | `FLAGGED_FILE_EXTENSION`   | warning  | Flagged file extensions found                       |
-| `DUPLICATE_XPI_ENTRY`      | warning  | Package contains duplicate entries                  |
 | `ALREADY_SIGNED`           | warning  | Already signed                                      |
 | `COINMINER_USAGE_DETECTED` | warning  | Firefox add-ons are not allowed to run coin miners. |
+| `DUPLICATE_XPI_ENTRY`      | error    | Package contains duplicate entries                  |
+| `INVALID_ZIPFILE`          | error    | Package is invalid. May contain invalid characters  |
```

Comment:
What's the difference between invalid and bad zipfile?

Reply:
bad zipfile seems to be associated to a zip file that is so corrupt that we are unable to open it (at least that is what I gather from the message description)

Reply:
@willdurand I looked a bit more into what would make BAD_ZIPFILE to be reported and how the error flows from yauzl to addons-linter:

- we recognize it based on the yauzl error message we expect to receive (See `ZIP_LIB_CORRUPT_FILE_ERROR` here https://github.com/mozilla/addons-linter/blob/c5975adb97b5f3c8aabbe8140f3973ffa390f1cf/src/const.js#L220-L221)

- we get an error that includes the ZIP_LIB_CORRUPT_FILE_ERROR string from yauzl's `open` callback parameter (unlike the other yauzl errors that we are now going to handle with the changes in this PR, which are received through 'error' events emitted on the zipfile instance, after it has been at least open successfully)

And so, from my perspective BAD_ZIPFILE, DUPLICATE_XPI_ENTRY and the new INVALID_ZIPFILE looks all like different error conditions (and seems reasonable to use separate error codes):
  - BAD_ZIPFILE: for zipfiles that are corrupted and can't be inspected by the linter
  - DUPLICATE_XPI_ENTRu: for zipfiles that can be opened just fine but are considered invalid because there are multiple entries with the exact same filename
  - INVALID_ZIPFILE: for zipfile that can be opened just fine, but some entries use characters that are considered invalid (and Firefox would refuse to load the zip file, even if the file isn't completely corrupted from a linter perspective and the linter was able to inspect the package)

Nevertheless, INVALID_ZIPFILE doesn't look like a good rule code name, it sounds too similar to BAD_ZIPFILE, and so I think that in this PR I should change the new rule name to something that would make it easier to don't confuse differetn errors between them.

I'm going to rename the new one into INVALID_XPI_ENTRY (insted of INVALID_ZIPFILE), because I think that we can assume that the "error" events emitted by yauzl on the zipfile instance after it has been opened successfully should always (or often enough) be related to invalid entries (instead of the entire zip file to be corrupted and the linter being unable to look into its entries).

Reply:
Excellent, thanks!